Updated Assessment

finneyj2 opened the pull request:
Made advised corrections to the categories Project Management with GitHub, Continuous Integration with Travis, and Foundations of Software Engineering. Mostly made grammar corrections, but one sentence had to be partially rewritten.

Commit:
made some corrections

Commit:
made some more corrections

Commit:
fixed third party

Commit:
made more corrections

finneyj2 committed:
add link

finneyj2 committed:
added link correctly this time

finneyj2 committed:
changed portion of pull request, may change again

finneyj2 committed:
changed portion of pull request -G, again

finneyj2 committed:
fixed pull request for hopefully the final time

finneyj2 committed:
trying to fix spacing for travis

Michionlion commented on assessment.md:
```diff
@@ -119,56 +119,57 @@ when creating python programs.
-  * I = Pushed to master or non-descriptive commits
-  * A = Pushed to correct repository with descriptive commits
+  * N = GitHub flow model unused or no commits to the Github repository
+  * I = Pushed to master or non-descriptive commits to the Github repository
```
These two should end with `made to the GitHub repository`

Michionlion commented on assessment.md:
```diff
@@ -119,56 +119,57 @@ when creating python programs.
-  * A = Pushed to correct repository with descriptive commits
+  * N = GitHub flow model unused or no commits to the Github repository
+  * I = Pushed to master or non-descriptive commits to the Github repository
+  * A = Pushed to correct repository with descriptive commits to the Github
```
This repeats the word repository unnecessarily, and should be reworded. There shouldn't be a need to append `to the repository` on the end of each of these sentences.

Lancasterwu replied:
Does it look better now?

Michionlion replied:
The line that was the problem is line 124 and 125 -- however this change is good as well!

Michionlion commented on assessment.md:
```diff
@@ -119,56 +119,57 @@ when creating python programs.
   (e.g having a feature branch) along with descriptive commits
   * E = Use of multiple useful branches/forks demonstrated appropriately
-    containing multiple coherent and descriptive commits
+    containing multiple consistent and descriptive commits
```
Coherent and consistent do not mean the same thing -> is the change in meaning acceptable?

I would argue that coherent commits are incredibly necessary.

finneyj2 replied:
I no longer have the paper with me, but if I remember correctly this change was suggested by @gkapfham.

Michionlion commented on assessment.md:
```diff
@@ -119,56 +119,57 @@ when creating python programs.
   * G = Merging branches after extensive communication
   * E = Merging branches with well-documented, detailed code and
-    extensive communication
+    extensive team communication
```
Communication is not necessarily in teams here... is there some justification for this change?

Michionlion commented on assessment.md:
```diff
@@ -119,56 +119,57 @@ when creating python programs.
+    extensive team communication
 * Creating and reviewing a pull request
-  * N = No PR made
+  * N = No PR(Pull Request) made
```
A space is needed before `(Pull Request)`. I would recommend simply replacing all instances of `PR` with `Pull Request` instead of using this style.

Lancasterwu replied:
Fixed

Michionlion commented on assessment.md:
```diff
@@ -119,56 +119,57 @@ when creating python programs.
     proposed changes by a person
   * G = Tags any related issues it is trying to close, describes most proposed changes,
-    discusses results of PR with Travis CI
+    and passes Travis CI
```
This is about creating and reviewing the PR, not creating code that passes Travis CI. The old wording was inadequate, but the new wording doesn't seem better.

Lancasterwu replied:
Fixed

Michionlion commented on assessment.md:
```diff
@@ -119,56 +119,57 @@ when creating python programs.
   * A = Commit messages are sensible but not descriptive
-  * G = Commit messages are both clear, and moderately descriptive
-  * E = Commit messages are clear, concise and descriptive
+  * G = Commit messages are clear, concise and moderately descriptive
```
The addition of concise here, in my opinion, makes the difference between level `G` and `E` too small. I think the original wording is better. This is of course the choice of the students involved, however.

Michionlion commented on assessment.md:
```diff
@@ -119,56 +119,57 @@ when creating python programs.
   * N = Fails to perform any setup and configuration of Travis CI
   * I = Sets up and configures Travis CI which performs no checks
-  * A = Sets up and configures Travis CI without performing checks on less
+  * A = Sets up and configures Travis CI that performs checks on less
```
Wait.... this inverts the meaning of the sentence, and seems that setting up checks on less relevant project deliverables is not as desirable? This is not clear.

Michionlion commented on assessment.md:
```diff
@@ -180,12 +181,12 @@ when creating python programs.
   * E = Performs a release through Travis CI with signed commits
     and signed deliverables (e.g md5 hash)
-* Create third-party integrations with Travis CI (e.g., codecov.io)
+* Create third-party integrations with Travis CI (e.g., [codecov](codecov.io)
```
Missing a closing `)`.

Michionlion commented on assessment.md:
```diff
@@ -202,30 +203,33 @@ when creating python programs.
 * Requirements engineering
-  * N = No effort to document, define, or maintain requirements present
+  * N = No effort to document, define, or maintain requirements present is
+  evident
```
`is evident` is redundant, `present` implies the same idea.

Michionlion commented on assessment.md:
```diff
@@ -202,30 +203,33 @@ when creating python programs.
   * N = There is no consideration to the structure or organization of the project
   * I = There are small amounts of structure to the system
-  * A = There is a blueprint for the software system that lacks completion
+  * A = There is a blueprint for the software system that is not complete
```
Possibly should be reworded to use `incomplete blueprint`.

Lancasterwu replied:
Fixed

Michionlion commented on assessment.md:
```diff
@@ -202,30 +203,33 @@ when creating python programs.
   * N = No attempt to create a design
   * I = Initial steps are taken to conceptualize, frame, and implement systems
-  * A = Some software design is visible to the point where a software solution looks
+  * A = Some software design is visible to the point where a software solution
```
This seems to have been reformatted weirdly.

Michionlion reviewed (changes requested):
Many small issues to address.

Additionally, there a quite a few language changes that modify the meaning of the levels that will probably need approval from some student leadership.

finneyj2 committed:
fixed formatting and parentheses

Lancasterwu committed:
Fixed `made to the GitHub repository`

Lancasterwu committed:
Fixed request repeat use of `GitHub repo`

Lancasterwu committed:
Fixed requested `or PR`

Lancasterwu committed:
Fixed requested "incomplete"

Michionlion commented:
@finneyj2 @Lancasterwu can both of you try to address the issues raised here so that this can be merged soon?

finneyj2 commented:
@Michionlion I no longer have the paper with me, but if I remember correctly, most of the changes that I implemented were suggested by @gkapfham.  Possibly all of them that are being questioned in this pull request. I think I made earlier errors and typos that I addressed earlier.
